src/sparseml/transformers/utils/sparse_model.py

```diff
@@ -261,29 +310,16 @@ def text_generation_from_pretrained(
         :return: the created model for text generation
         """
 
-        def skip(*args, **kwargs):
-            pass
-
-        # Skip the initializer step. This accelerates the loading
-        # of the models, especially for the quantized models
-        torch.nn.init.kaiming_uniform_ = skip
-        torch.nn.init.uniform_ = skip
-        torch.nn.init.normal_ = skip
-
-        model = AutoModelForCausalLM.from_pretrained(
+        model = SparseAutoModelForCausalLM.from_pretrained(
             model_name_or_path,
             torch_dtype=torch_dtype,
             trust_remote_code=trust_remote_code,
+            recipe=recipe,
```

Comment:
Theres a lot of logic that goes into resolving the recipe. Could we add in a lifecycle list to the comments of this fn to summarize it?

Reply:
I've broken down the logic into smaller functions. Hopefully it helps.